src/components/structures/MatrixChat.js

```diff
@@ -92,8 +118,8 @@ module.exports = React.createClass({
     componentDidMount: function() {
         this._autoRegisterAsGuest = false;
         if (this.props.enableGuest) {
-            if (!this.props.config || !this.props.config.default_hs_url) {
-                console.error("Cannot enable guest access: No supplied config prop for HS/IS URLs");
+            if (!this.getCurrentHsUrl() || !this.getCurrentIsUrl()) {
+                console.error("Cannot enable guest access: can't determine HS/IS URLs to use");
```

Comment:
Do we *really* need an IS URL specified to register as a guest? I think you'll find we don't.